src/Rector/v11/v4/ProvideCObjViaMethodRector.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Ssch\TYPO3Rector\Rector\v11\v4;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\Name\FullyQualified;
+use PhpParser\Node\Stmt\Class_;
+use PhpParser\Node\Stmt\ClassMethod;
+use PhpParser\Node\Stmt\Nop;
+use PhpParser\Node\Stmt\Property;
+use PHPStan\Type\ObjectType;
+use Rector\Core\Rector\AbstractRector;
+use Symplify\Astral\ValueObject\NodeBuilder\MethodBuilder;
+use Symplify\Astral\ValueObject\NodeBuilder\ParamBuilder;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @changelog https://docs.typo3.org/c/typo3/cms-core/master/en-us/Changelog/11.4/Deprecation-94956-PublicCObj.html
+ * @see \Ssch\TYPO3Rector\Tests\Rector\v11\v4\ProvideCObjViaMethodRector\ProvideCObjViaMethodRectorTest
+ */
+final class ProvideCObjViaMethodRector extends AbstractRector
+{
+    /**
+     * @var string
+     */
+    private const COBJ = 'cObj';
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [Class_::class];
+    }
+
+    /**
+     * @param Class_ $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if ($this->isObjectType($node, new ObjectType('TYPO3\CMS\Frontend\Plugin\AbstractPlugin'))
+            || $this->isObjectType($node, new ObjectType('TYPO3\CMS\Extbase\Mvc\Controller\ActionController'))
+        ) {
+            return null;
+        }
+
+        $cObjProperty = $node->getProperty(self::COBJ);
+        if (! $cObjProperty instanceof Property) {
+            return null;
+        }
+
+        if ($cObjProperty->isPublic()) {
+            // TODO: add handling to make it protected
+        }
+
+        $classMethod = $node->getMethod('setContentObjectRenderer');
+        if ($classMethod instanceof ClassMethod) {
+            return null;
+        }
+
+        $this->addSetContentObjectRendererMethod($node);
```

Comment:
Here the modified class node should be returned, as you're adding an item to array.
That's my rough guess :)